packages/virtual-list/src/vaadin-virtual-list-mixin.js

```diff
@@ -116,18 +125,34 @@ export const VirtualListMixin = (superClass) =>
       return [...Array(count)].map(() => document.createElement('div'));
     }
 
+    /** @private */
+    __updateAria() {
+      this.role = 'list';
+    }
+
     /** @private */
     __updateElement(el, index) {
+      const item = this.items[index];
+      el.ariaSetSize = String(this.items.length);
+      el.ariaPosInSet = String(index + 1);
```

Comment:
It seems that aria properties are more commonly set using attributes in our codebase.

```suggestion
      el.setAttribute('aria-setsize', this.items.length);
      el.setAttribute('aria-posinset', index + 1);
```

Reply:
Why shouldn't type-safe DOM API be preferred when available since they are easier to use, more readable, and less error-prone than setAttribute?

Reply:
Good point about type-safety.